src/java/htsjdk/samtools/util/BlockCompressedInputStream.java

```diff
@@ -41,25 +41,27 @@
 import java.nio.ByteOrder;
 import java.util.Arrays;
 
-/*
+/**
  * Utility class for reading BGZF block compressed files.  The caller can treat this file like any other InputStream.
  * It probably is not necessary to wrap this stream in a buffering stream, because there is internal buffering.
  * The advantage of BGZF over conventional GZip format is that BGZF allows for seeking without having to read the
- * entire file up to the location being sought.  Note that seeking is only possible if the ctor(File) is used.
+ * entire file up to the location being sought.  Note that seeking is only possible if the input stream is seekable.
+ * 
+ * Note that this implementation is not synchronized. If multiple threads access an instance concurrently, it must be synchronized externally. 
  *
  * c.f. http://samtools.sourceforge.net/SAM1.pdf for details of BGZF format
+ * 
+ * 
  */
 public class BlockCompressedInputStream extends InputStream implements LocationAware {
```

Comment:
hadoop-bam is still failing for me when i use this PR's code. The stream is synchronized and it does not fail on htsjdk 2.2.2 code. 
```
Caused by: java.io.IOException: Invalid file pointer: 1049
	at htsjdk.samtools.util.BlockCompressedInputStream.seek(BlockCompressedInputStream.java:308)
	at org.seqdoop.hadoop_bam.BAMRecordReader.initialize(BAMRecordReader.java:153)
	at org.apache.spark.rdd.NewHadoopRDD$$anon$1.<init>(NewHadoopRDD.scala:133)
	at org.apache.spark.rdd.NewHadoopRDD.compute(NewHadoopRDD.scala:104)
	at org.apache.spark.rdd.NewHadoopRDD.compute(NewHadoopRDD.scala:66)
	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:297)
```